components/scream/src/physics/shoc/shoc_functions.hpp

```diff
@@ -611,6 +659,65 @@ struct Functions
     const uview_1d<Spack>&       wqls_sec,
     const uview_1d<Spack>&       brunt,
     const uview_1d<Spack>&       isotropy);
+#else
+  static void shoc_main_internal(
```

Comment:
Maybe we could give these functions two different names, like `shoc_main_monolithic` and `shoc_main_multiple_kernels`, or something.